tests/test_packages/test_contracts/test_erc1155/test_contract.py

```diff
@@ -300,109 +305,100 @@ def sign_send_verify_handle_transaction(self, tx: Dict[str, str], sender_crypto)
-            [
-                key in receipt
-                for key in [
-                    "height",
-                    "txhash",
-                    "data",
-                    "raw_log",
-                    "logs",
-                    "gas_wanted",
-                    "gas_used",
-                ]
-            ]
-        )
-
-    @pytest.mark.skip
-    @pytest.mark.integration
-    @pytest.mark.ledger
-    def test_cosmwasm_contract_deploy_and_interact(self, erc1155_contract):
-        """Test cosmwasm contract deploy and interact."""
-        # Deploy contract
-        contract, contract_address = erc1155_contract
-        tx = contract.get_deploy_transaction(
+        tx_hash = self.ledger_api.send_signed_transaction(signed_tx)
+        tx_receipt = self.ledger_api.get_transaction_receipt(tx_hash)
+        assert len(tx_receipt) == 8
+        assert self.ledger_api.is_transaction_settled(tx_receipt), tx_receipt["raw_log"]
+
+    def set_contract(self):
+        """Set contract."""
+        directory = Path(ROOT_DIR, "packages", "fetchai", "contracts", "erc1155")
+        configuration = load_component_configuration(ComponentType.CONTRACT, directory)
+        configuration._directory = directory
+        configuration = cast(ContractConfig, configuration)
+
+        if str(configuration.public_id) not in contract_registry.specs:
+            # load contract into sys modules
+            Contract.from_config(configuration)
+
+        self.contract = contract_registry.make(str(configuration.public_id))
+
+    def deploy_contract(self):
+        """Deploy contract."""
+        tx = self.contract.get_deploy_transaction(
             ledger_api=self.ledger_api,
             deployer_address=self.deployer_crypto.address,
-            gas=900000,
+            gas=2000000,
         )
         assert len(tx) == 6
-        self.sign_send_verify_deploy_init_transaction(tx, self.deployer_crypto)
-        code_id = contract.get_last_code_id(self.ledger_api)
+        signed_tx = self.deployer_crypto.sign_transaction(tx)
+        tx_hash = self.ledger_api.send_signed_transaction(signed_tx)
+        tx_receipt = self.ledger_api.get_transaction_receipt(tx_hash)
+        assert len(tx_receipt) == 9
+        assert self.ledger_api.is_transaction_settled(tx_receipt), tx_receipt["raw_log"]
+
+        code_id = self.ledger_api.get_code_id(tx_receipt)
+
+        assert code_id is not None
+        assert code_id == self.ledger_api.get_last_code_id()
+        self.code_id = code_id
 
         # Init contract
-        tx = self.ledger_api.get_init_transaction(
-            self.deployer_crypto.address,
-            code_id,
+        tx = self.contract.get_deploy_transaction(
+            ledger_api=self.ledger_api,
+            deployer_address=self.deployer_crypto.address,
+            code_id=self.code_id,
             init_msg={},
             tx_fee=0,
             amount=0,
             label="ERC1155",
+            gas=1000000,
```

Comment:
Was the gas too low before? Is this that contributed to fixing #2008?

Reply:
Yes the gas was too low. But the whole PR makes changes required for the test to pass. Not just the gas